### DISCUSSION TITLE
How to get Avro Schema?

### DISCUSSION THREAD

**Discussion by kakoc** [gklijs/schema_registry_converter#66] [Q&A]
Hi

I didn't find any example how I can retrieve a schema(AvroSchema type) for some topic.
**to_avro_schema**,  **get_schema_and_id**, **get_schema** are all private.
Does it make sense to make some of them public? Or probably I missed smth and it's possible now to retrieve Avro Schema using this crate?

**Comment by gklijs** [comment]
As of now this library's main purpose is to do the conversion of data using Schema Registry, you can provide a schema when doing the conversion and it will be registered if needed.

I do consider splitting this library, and have a seperate Schema Registry Client, that allows for getting just the schema and such, independent of (de) serializing data. I'm not entirely, sure if that would be worth the effort.

**Comment by gklijs** [comment]
I gave it some thought and I'm thinking about adding a feature like `sr-client` that would give you a clean API to get/set schema's by SubjectNameStrategy or `subject` where you could get the latest, or a specific version. Would that be helpful?

**Reply by ultrabug** [reply]
Why couldn't we just expose the cache from the AvroDecoder and schema id on the DecodeResult struct instead?

I must admit I hoped I found it myself there but didn't, yet that's some information that exist AFAIK

**Reply by ultrabug** [reply]
See, I asked for the same stuff here https://github.com/gklijs/schema_registry_converter/discussions/62#discussioncomment-1324532

**Reply by gklijs** [reply]
It's already kind of intermediate, and than you would first need to get it in the cache.
